python/packages/nisar/workflows/nisar_l0b_dm2_to_dbf.py

```diff
@@ -132,6 +157,137 @@ def create_echo_dataset_h5(fid_in, fid_out, band_path, txrx_pol,
     return dset_prod
 
 
+def _copy_datasets_truncated(
+        fid_in, fid_out, band_path, pol, pulse_slice, logger, keys):
+    """
+    Copy truncated version of non-echo pulse-dependent datasets under
+    either tx or rx group from input HDF5 to output one given desired
+    input pulse slice.
+    The type of tx or rx is determined by the size of pol.
+    If single char then it is tx group and if two-char str then
+    it will be dataset under rx group
+
+    Paramaters
+    ----------
+    fid_in : h5py.File
+        File-like object for input HDF5 L0B product
+    fid_out : h5py.File
+        File-like object for output HDF5 L0B product
+    band_path : str
+        Frequency band path in L0B HDF5 product.
+    pol : str
+        Either signle-char tx pol or two-char TxRx polarization
+        of the new echo product
+    pulse_slice : slice
+        Desired pulse slice for input HDF5 file.
+    logger: logging.Logger
+    keys: sequence of str
+        Desired keys under RX group to be copied as long as
+        they exist in input hdf5.
+
+    """
+    path = os.path.join(band_path, f'tx{pol[0]}/')
+    if len(pol) == 2:
+        path += f'rx{pol[1]}/'
+    # output group
+    grp_out = fid_out[path]
+    for name in set(keys):
+        try:
+            dset_in = fid_in[path + name]
+        except KeyError as err:
+            logger.warning(
+                f'Missing dataset "{path + name}" in input L0B!'
+                f' Detailed Error -> "{err}"'
+            )
+            continue
+        else:
+            data_in = dset_in[pulse_slice.start:pulse_slice.stop]
```

Comment:
```suggestion
            data_in = dset_in[pulse_slice]
```
Simpler syntax and doesn't ignore possible stride.